### PR TITLE
Allow compiling LLVM backend in library mode

### PR DESCRIPTION
**Pull request by ehildenb**
This enables compiling the LLVM backend in library mode by adding a new KompileBackend called LLVM_LIB.

- Add way to add `--llvm-kompile-type c` to arguments of LLVM compilation via `linkable` flag.
- Remove `JAVA` kompile backend.

This makes it possible to compile the LLVM backend at the same time as the booster in KEVM.

### CODE REVIEW

**Comment by ehildenb** [comment]
Not sure if this is the best solution, or if it's better to have a separate `LLVM_MODE` enum. But it seems you can't both have normal compilation and library compilation, because they write to the same file. Let me know if we should do this differently.

File: src/pyk/ktool/kompile.py [hunk 1]
```diff
@@ -37,6 +36,7 @@ def kompile(
     opt_level: Optional[int] = None,
     ccopts: Iterable[str] = (),
     no_llvm_kompile: bool = False,
+    linkable: bool = False,
```
Comment (tothtamas28) [inline src/pyk/ktool/kompile.py:39]: This should be `llvm_kompile_type: Optional[LlvmKompileType]` where

```
class LlvmKompileType(Enum):
    MAIN = 'main'
    SEARCH = 'search'
    LIBRARY = 'library'
    STATIC = 'static'
    PYTHON = 'python'
    C = 'c'
```

Comment (ehildenb) [reply src/pyk/ktool/kompile.py]: Done.

**Review by tothtamas28 — APPROVED** [review]

File: src/pyk/ktool/kompile.py [hunk 2]
```diff
@@ -56,6 +65,7 @@ def kompile(
         _check_backend_param(opt_level is None, 'opt_level', backend)
         _check_backend_param(not list(ccopts), 'ccopts', backend)
         _check_backend_param(not no_llvm_kompile, 'no_llvm_kompile', backend)
+        _check_backend_param(llvm_kompile_type is not None, 'linkable', backend)
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply src/pyk/ktool/kompile.py]: Done.

File: src/pyk/ktool/kompile.py [hunk 3]
```diff
@@ -37,6 +45,7 @@ def kompile(
     opt_level: Optional[int] = None,
     ccopts: Iterable[str] = (),
     no_llvm_kompile: bool = False,
+    llvm_kompile_type: Optional[LLVMKompileType] = None,
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply src/pyk/ktool/kompile.py:48]: Done.

**Pull request merged**